tests/scripts/test_task.sh

```diff
@@ -10,14 +10,22 @@ cleanup() {
 trap cleanup EXIT
 
 sync_task="pulp_file.app.tasks.synchronizing.synchronize"
-count=$(pulp_cli task list --name $sync_task --state canceled | jq -r .count)
+count=$(pulp_cli task list --name $sync_task --state canceled | jq -r length)
 
 pulp_cli file remote create --name "cli_test_file_remote" \
   --url "https://fixtures.pulpproject.org/file-large/PULP_MANIFEST"
 pulp_cli file repository create --name "cli_test_file_repository" --remote "cli_test_file_remote"
 
+# Test canceling a task
 task=$(pulp_cli file repository sync --background --name "cli_test_file_repository" 2>&1 | egrep -o "/pulp/api/v3/tasks/[-[:xdigit:]]*/")
 pulp_cli task cancel --href "$task"
-test $(pulp_cli task list --name $sync_task --state canceled | jq -r .count) -eq $(($count + 1))
+test $(pulp_cli task list --name $sync_task --state canceled | jq -r length) -eq $(($count + 1))
+test $(pulp_cli task show --href $task | jq -r '.state') = "canceled"
+
+# Test waiting for a task
+task=$(pulp_cli file repository sync --background --name "cli_test_file_repository" 2>&1 | egrep -o "/pulp/api/v3/tasks/[-[:xdigit:]]*/")
+test $(pulp_cli task show --wait --href $task | jq -r '.state') = "completed"
```

Comment:
Not quite sure if this is a good idea, because this sync task takes forever...

Reply:
I tried testing this on my system but I am getting a strange error:

```
sync_task="pulp_file.app.tasks.synchronizing.synchronize"                                                                                                                           
count=$(pulp_cli task list --name $sync_task --state canceled | jq -r length)                                                                                                       
parse error: Invalid numeric literal at line 1, column 2                                                                                                                            
```

Reply:
Actually, I had my format set to "yaml". Need to figure out how to handle settings for tests.

I ran this locally and the entire test took less than one min so I think that's ok.